kart/tabular/working_copy/base.py

```diff
@@ -454,13 +454,43 @@ def diff_repo_to_working_copy(
             repo_diff.prune()
             return repo_diff
 
+    def diff_table_repo_to_working_copy(self, ds_diff):
+        """
+        Generates a diff between a working copy DB and the underlying repository tree,
+        for a single dataset.
+        """
```

Comment:
this no longer does what the docstring says - it just produces a RepoDiff from a single dataset diff. Probably you can remove the function and just do that in your command.

I'm not sure what `repo_diff.prune()` does - have a look and decide whether you actually need to call it or not